src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -303,6 +303,7 @@ void ThreadStatus::logToQueryThreadLog(QueryThreadLog & thread_log)
     QueryThreadLogElement elem;
 
     elem.event_time = time(nullptr);
+    elem.event_time_microseconds = getCurrentTimeMicroseconds();
```

Comment:
@tavplubix , is that because of the way I construct the time differently for `event_time` and `event_time_microseconds`? 
Could constructing them from timespec directly yield better results?
```c
    struct timespec ts;
    clock_gettime(clock_type, &ts);
    elem.event_time = ts.tv_sec
    elem.event_time_microseconds = UInt64((ts.tv_sec * 1000000LL) + (ts.tv_nsec / 1000));
```
Maybe I missed the point here, do let me know if there's a better way? Thanks!

Reply:
Yes, we should call `clock_gettime(...)` (or similar function) once and and use the result for both `event_time` and `event_time_microseconds`. Otherwise the values may be unexpected:
```
/// Current time is 2020-09-11 10:08:17.999999
elem.event_time = time(nullptr);    /// Returns 2020-09-11 10:08:17
/// ... and 1ms later
elem.event_time_microseconds = getCurrentTimeMicroseconds();    /// Returns 2020-09-11 10:08:18.000000
/// Now elem.event_time and elem.event_time_microseconds have different seconds part
```
And `SELECT if(dateDiff('second', toDateTime(time_with_microseconds), toDateTime(time)) = 0, 'ok', 'fail');` will return `'fail'`

Reply:
sure, that makes sense, I'll make the change.